tutorials/cloudbuild-angular-universal/index.md

```diff
@@ -0,0 +1,301 @@
+gcloud compute forwarding-rules create http-content-rule \
+--address angular-app-ip \
+--global \
+--target-http-proxy http-lb-proxy \
+--ports 80
+```
+## Create the Cloudbuild file and add it to the git repsoitory
+1. Give the cloudbuild cloud storage admin access.
+
+```bash
+CLOUD_BUILD_ACCOUNT=$(gcloud projects get-iam-policy $PROJECT --filter="(bindings.role:roles/cloudbuild)"  --flatten="bindings[].members" --format="value(bindings.members[])")
+gcloud projects add-iam-policy-binding $PROJECT   --member $CLOUD_BUILD_ACCOUNT  --role roles/storage.admin
+```
+2. Create the ```cloudbuild.yaml``` file.
+
+```bash
+cat <<CLOUDBUILD_FILE>cloudbuild.yaml
+steps:
+- id: install_packages
+  name: 'gcr.io/cloud-builders/npm'
+  args:
+  - 'install'
+- id: prerender_browser_files
+  name: 'gcr.io/cloud-builders/npm'
+  args:
+  - 'run'
+  - 'build:prerender'
+  waitFor:
+  - install_packages
+- id: copy_prerendered_files
+  name: 'gcr.io/cloud-builders/gsutil'
+  args: ['cp','-r','dist/browser/*', '\${_ANGULAR_APP_BUCKET_PATH}']
+  waitFor:
+  - prerender_browser_files
+- id: set_website_configuration
+  name: 'gcr.io/cloud-builders/gsutil'
+  args: ['web', 'set', '-m', 'index.html','\${_ANGULAR_APP_BUCKET_PATH}']
+  waitFor:
+  - copy_prerendered_files
+- id: set_permissions_for_website_files
+  name: 'gcr.io/cloud-builders/gsutil'
+  args: ['acl','ch','-u','AllUsers:R','-r', '\${_ANGULAR_APP_BUCKET_PATH}']
+  waitFor:
+  - copy_prerendered_files
+CLOUDBUILD_FILE
+```
+
+3. Add and commit ```cloudbuild.yaml``` to the git repository.
+```
+git add cloudbuild.yaml && git commit -m "add cloudbuild.yaml"
+```
+### Create a Build Trigger that will build, test and deploy your application to the Cloud CDN
+You can create a trigger on the [build triggers page](https://console.cloud.google.com/cloud-build/triggers) of the GCP Console by following these steps:
+
+1. Click **Create Trigger** OR **Add Trigger** button
+1. Select **Cloud Source Repository** Click **Continue**.
+1. Select ```tour-of-heroes-universal``` and click **Continue**.
+1. Enter ```angular-universal-tour``` for **Name.**
+1. Under **Trigger type** select "Tag".
+1. Under **Build configuration** select ```cloudbuild.yaml```
+1. Under **Substitution variables** press **+ Add item**
+1. In the **Variable** field enter```_ANGULAR_APP_BUCKET_PATH``` and in **Value** enter ```gs://[PROJECT]-angular-app``` where ```[PROJECT]``` is the name of your project
+1. Click **Create Trigger**
+
+### Add your tour-of-heros cloud source repository as a remote repository with the name 'google'
+
+1.  (**OPTIONAL**: Only if not running in Cloud Shell) Set up your google credentials for git.
+```
+gcloud init && git config --global credential.https://source.developers.google.com.helper gcloud.sh
+```
+
+2.  Add the google cloud repo as a remote.
+```
+git remote add google \
+https://source.developers.google.com/p/$PROJECT/r/tour-of-heros-universal
+```
+
+## Run the Build Trigger and Deploy the Application
```

Comment:
'Run the Build Trigger and Deploy the Application'
->
'Run the build trigger and deploy the application'